components/series/stackedBar.js

```diff
@@ -6,77 +6,39 @@
         var stackedBar = function(selection) {
             var container;
 
-            // takes an object with values associated with each property, and
-            // converts it into an array of values. Each value has the xValue associated
-            // with it.
-            //
-            // For example, this object:
-            //
-            // obj = { county: 'North Tyneside', labour: 23, conservative: 55 }
-            //
-            // becomes this ...
-            //
-            // [
-            //   { xValue: 'North Tyneside', name: 'labour', previousValue: 0, currentValue: 23},
-            //   { xValue: 'North Tyneside', name: 'conservative', previousValue: 23, currentValue: 78},
-            // ]
-            function objectDatapointToArray(obj) {
-                var values = [];
-                var yTotal = 0;
-                var xVal = obj[stackedBar.xValueKey.value];
-                for (var propertyName in obj) {
-                    if (obj.hasOwnProperty(propertyName) && propertyName !== stackedBar.xValueKey.value) {
-                        var previous = yTotal;
-                        yTotal += Number(obj[propertyName]);
-                        values.push({
-                            'name': propertyName,
-                            'previousValue': previous,
-                            'currentValue': yTotal,
-                            'xValue': xVal
-                        });
-                    }
-                }
-                return values;
-            }
-
             selection.each(function(data) {
 
                 // add a 'root' g element on the first enter selection. This ensures
                 // that it is just added once
                 container = d3.select(this);
 
-                var keyFunction = function(d) {
-                    return d[stackedBar.xValueKey.value];
-                };
-                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', data, keyFunction);
+                // If the user has already set stacked data then we don't need to re-compute it.
+                var layers = stackedBar.stackedData.value;
+                if (typeof layers === 'undefined' || layers.length === 0) {
```

Comment:
JS stylistic things... 
* Prefer ```!= null``` to anything involving undefined. Generally ```null``` and ```undefined``` are conflated as being the same value, using the bang equals null approach will match either (notice the use of double rather than triple equality). An exception to the rule can be for testing for the existence of arguments but testing the length of the ```arguments``` object is normally more appropriate.
* Either use an empty array to represent empty and operate on it as if it were the real thing, or choose null to represent empty and special case it. I would avoid using an empty array as a special case.